src/app/configuration/configuration.component.ts

```diff
@@ -0,0 +1,114 @@
+      planet_type: [ '', Validators.required ],
+      local_domain: [ '', Validators.required ],
+      name: [ '', Validators.required ],
+      parent_domain: [ '', Validators.required ],
+      preferred_lang: [ '', Validators.required ],
+      code: [ '', Validators.required ]
+    });
+    this.contactFormGroup = this._formBuilder.group({
+      firstName: [ '', Validators.required ],
+      lastName: [ '', Validators.required ],
+      middleName: [ '' ],
+      email: [ '', Validators.required ],
+      phoneNumber: [ '', Validators.required ]
+    });
+    this.getNationList();
+  }
+
+  getNationList() {
+    this.couchService.get('nations/_all_docs?include_docs=true')
+      .subscribe((data) => {
+        for (let i = 0; i < data.rows.length; i++) {
+          if (data.rows[i].doc['_id'].indexOf('_design') === -1) {
+            this.nations.push(data.rows[i].doc);
+          }
+        }
+      }, (error) => this.message = 'There was a problem getting NationList');
+  }
+
+  onChange(selectedValue: string) {
+    this.select = selectedValue;
+    if (this.select === 'community') {
+      this.configurationFormGroup = this._formBuilder.group({
+        planet_type: [ 'community', Validators.required ],
+        local_domain: [ document.baseURI, Validators.required ],
+        parent_domain: [ '', Validators.required ],
+        name: [ '', Validators.required ],
+        preferred_lang: [ '', Validators.required ],
+        code: [ '', Validators.required ]
+      });
+    } else {
+      this.configurationFormGroup = this._formBuilder.group({
+        planet_type: [ 'nation', Validators.required ],
+        name: [ '', Validators.required ],
+        parent_domain: [ 'nbs.ole.org:5997', Validators.required ],
+        local_domain: [ document.baseURI, Validators.required ],
+        preferred_lang: [ '', Validators.required ],
+        code: [ '', Validators.required ]
+      });
+    }
+  }
+
+  checkPassword() {
+    if (this.loginForm.value.password === this.loginForm.value.confirmPassword) {
+      this.stepper.selectedIndex = 1;
+      this.message = '';
+    } else {
+      this.message = 'Passwords do not match';
+    }
+  }
+
+  onSubmitConfiguration() {
+    if (this.loginForm.valid && this.configurationFormGroup.valid && this.contactFormGroup.valid) {
+      this.couchService.put('_node/nonode@nohost/_config/admins/' + this.loginForm.value.username, this.loginForm.value.password)
+        .subscribe((data) => {
+          const config = Object.assign({}, this.configurationFormGroup.value, this.contactFormGroup.value);
+          this.couchService.post('configurations', config).subscribe(() => {
+          this.router.navigate([ 'login' ]);
+          }, (err) => {
+            // Connect to an error display component to show user that an error has occurred
+            console.log(err);
+          });
+        }, (error) => (error));
+    }
+  }
+
+  cancel() {
+    this.router.navigate([ 'login' ]);
```

Comment:
cancel can not take to login. Until config is setup none of the page should be accessible